app/queues/dns/create-record-worker.server.ts

```diff
@@ -0,0 +1,145 @@
+import { UnrecoverableError, Worker } from 'bullmq';
+import { RecordStatus } from '@prisma/client';
+
+import { redis } from '~/lib/redis.server';
+import { prisma } from '~/db.server';
+import logger from '~/lib/logger.server';
+import { setMonthsFromNow } from '~/lib/domains.server';
+import { createRecord, getChangeStatus } from '~/lib/dns.server';
+
+import type { Record } from '@prisma/client';
+
+export const addDbRecordQueueName = 'add-db-record';
+export const createDnsRecordQueueName = 'create-dns-record';
+export const checkDnsStatusQueueName = 'check-dns-status';
+export const syncDbStatusQueueName = 'sync-db-status';
+
+export interface JobRecord {
+  username: Record['username'];
+  type: Record['type'];
+  name: Record['name'];
+  value: Record['value'];
+}
+
+export const addDbRecordWorker = new Worker<JobRecord>(
+  addDbRecordQueueName,
+  async (job) => {
+    logger.info('Creating a record in DB');
+    const { username, type, name, value } = job.data;
+
+    const existingRecords = await prisma.record.count({
+      where: {
+        type,
+        name,
+        value,
+      },
+    });
+
+    if (existingRecords) {
+      throw new Error('Record already exists');
+    }
+
+    try {
+      const result = await prisma.record.create({
+        data: {
+          username,
+          type,
+          name,
+          value,
+          expiresAt: setMonthsFromNow(6),
+          status: RecordStatus.pending,
+        },
+      });
+      return result.id as Record['id'];
```

Comment:
```suggestion
      return result.id;
```
Not needed because of the above generic